test/SparqlAntlrParserTestHelpers.h

```diff
@@ -0,0 +1,123 @@
+// Copyright 2022, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Robin Textor-Falconi (textorr@informatik.uni-freiburg.de)
+
+#pragma once
+
+#include <gmock/gmock.h>
+
+#include "../src/parser/data/VarOrTerm.h"
+#include "../src/util/TypeTraits.h"
+
+// Not relevant for the actual test logic, but provides
+// human-readable output if a test fails.
+std::ostream& operator<<(std::ostream& out, const GraphTerm& graphTerm) {
+  std::visit(
+      [&]<typename T>(const T& object) {
+        if constexpr (ad_utility::isSimilar<T, Literal>) {
+          out << "Literal " << object.literal();
+        } else if constexpr (ad_utility::isSimilar<T, BlankNode>) {
+          out << "BlankNode generated: " << object.generated()
+              << ", label: " << object.label();
+        } else if constexpr (ad_utility::isSimilar<T, Iri>) {
+          out << "Iri " << object.iri();
+        } else {
+          static_assert(ad_utility::alwaysFalse<T>, "unknown type");
+        }
+      },
+      static_cast<const GraphTermBase&>(graphTerm));
+  return out;
+}
+
+// _____________________________________________________________________________
+
+std::ostream& operator<<(std::ostream& out, const VarOrTerm& varOrTerm) {
+  std::visit(
+      [&]<typename T>(const T& object) {
+        if constexpr (ad_utility::isSimilar<T, GraphTerm>) {
+          out << object;
+        } else if constexpr (ad_utility::isSimilar<T, Variable>) {
+          out << "Variable " << object.name();
+        } else {
+          static_assert(ad_utility::alwaysFalse<T>, "unknown type");
+        }
+      },
+      static_cast<const VarOrTermBase&>(varOrTerm));
+  return out;
+}
+
+// _____________________________________________________________________________
+
+template <typename, typename... Ts>
+struct LastT : LastT<Ts...> {};
+
+template <typename T>
+struct LastT<T> {
+  using type = T;
+};
+
+template <typename... Ts>
+using Last = typename LastT<Ts...>::type;
+
+template <typename T, typename...>
+struct FirstWrapper {

```

Comment:
Could be `template<....> struct FirstWrapper : std::type_identity<T> {};`

But not sure if this is better.